ramp-engine/ramp_engine/remote.py

```diff
@@ -0,0 +1,302 @@
+        ).result()
+        super().teardown()
+        self._client.close()
+
+    def _is_submission_finished(self):
+        """Status of the submission.
+
+        The submission was launched in a subprocess. Calling ``poll()`` will
+        indicate the status of this subprocess.
+        """
+        self.check_timeout()
+        return self._proc.done()
+
+    def check_timeout(self):
+        """Check the submission for timeout."""
+        if not hasattr(self, "_start_date"):
+            return
+        dt = (datetime.utcnow() - self._start_date).total_seconds()
+        if dt > self.timeout:
+            self._proc.cancel()
+            self.status = "timeout"
+            return True
+
+    @property
+    def timeout(self):
+        return self.config.get('timeout', 7200)
+
+    def launch_submission(self):
+        """Launch the submission.
+
+        Basically, it comes to run ``ramp_test_submission`` using the conda
+        environment given in the configuration. The submission is launched in
+        a subprocess to free to not lock the Python main process.
+        """
+        cmd_ramp = os.path.join(self._python_bin_path, 'ramp-test')
+        if self.status == 'running':
+            raise ValueError('Wait that the submission is processed before to '
+                             'launch a new one.')
+
+        self._log_dir = os.path.join(self.config['logs_dir'], self.submission)
+        # TODO: need to copy submission to the remote folder
+        if not self._is_local_cluster:
+            submission_dir = os.path.join(
+                self.config['submissions_dir'], self.submission
+            )
+            # remove remote submission and local prediction dir if it exists
+            self._client.submit(
+                shutil.rmtree,
+                submission_dir,
+                ignore_errors=True,
+                pure=False
+            ).result()
+            # Upload the submission dir to the remote machine
+            stream = _serialize_folder(submission_dir)
+            self._client.submit(
+                _deserialize_folder, stream, submission_dir, pure=False
+            ).result()
+            # If there is a training_output in the submission, remove it
+            self._client.submit(
+                _remove_link_or_dir,
+                os.path.join(submission_dir, 'training_output'),
+                pure=False
+            ).result()
+
+        self._proc = self._client.submit(
+            _conda_ramp_test_submission,
+            self.config,
+            self.submission,
+            cmd_ramp,
+            self._log_dir,
+            wait=True,
+            pure=False,
+        )
+
+        super().launch_submission()
+        self._start_date = datetime.utcnow()
+
+    def collect_results(self):
+        """Collect the results after that the submission is completed.
+
+        Be aware that calling ``collect_results()`` before that the submission
```

Comment:
we could maybe add a check for this? if the worker.status is `running` we don't collect the results. If yes this can be refactored later maybe

Reply:
Yes, that's what happens in the dispatcher, which handles when to call these methods depending on the worker status. However here this comment is still useful just to indicate the the call to `collect_results` is blocking.